nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -221,7 +240,11 @@ protected void initRepository() {
         coll.createIndex(Indexes.ascending(KEY_PROXY_TARGET_ID));
         coll.createIndex(Indexes.ascending(KEY_PROXY_VERSION_SERIES_ID));
         coll.createIndex(Indexes.ascending(KEY_READ_ACL));
-        coll.createIndex(Indexes.ascending(KEY_PARENT_ID, KEY_NAME));
+        IndexOptions parentNameIndexOptions = new IndexOptions();
+        if (!isNull(descriptor)) {
+            parentNameIndexOptions.unique(isTrue(descriptor.getChildNameUniqueConstraintEnabled()));
```

Comment:
`Boolean.TRUE.equals(...)` is enough, no need to import a helper for this. Also it makes it explicit that we're interested in `Boolean.TRUE` and not just the basic `true`.